### ISSUE TITLE
not an issue but....

### ISSUE THREAD

**Issue by phear46** [wpatoolkit/10-Hex-Generator#1]
Just wondering how many permutations exist with the -no01 option. im sure the answer is obvious but my maths is decidedly lacking!

Edit: just a thought, maybe a calculation of the following work, like crunch does. That would be a cool feature.

**Comment by wpatoolkit** [comment]
Without any rules there would be 289,254,654,976 (14^10) combinations (~3,471.06 GB).

If -no01 is used...
-there would be 183,780,015,460 combinations (or 63.5% of the keyspace) (~2,205.36 GB)

If -no01 and -risky are used...
-there would be 134,081,483,174 combinations (or 46.3% of the keyspace) (~1,608.98 GB)

**Comment by nd001** [comment]
Great work this wpatoolkit. Have you successfully managed to crack or find any BTHomeHub5 wireless keys? If so what options did you use to create your wordlist? 

I have a few BTHomeHub5's and will be trying to see if i can find the wireless keys in my wordlists.  

1609Gb is not that much these days. 2TB HDD's retail between £60 - 70. The time to write the file is not too bad either. Not sure how long it would take to crack though. On my laptop aircrack can check approx 1000keys per second against a wordlist. 

The problem is i reckon this could take 4.25 years if -no01 and -risky are used. Am i correct? 

Is there a quicker way? If i increased CPU / GPU power and could check 4000 keys per second, it would still take a year.

I suppose the only answer is more computing power.

**Comment by DarrenRainey** [comment]
Yeah I would like to see if there is away of generating a smaller keyspace for the BT Home Hubs
